### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by Sparsh1212**
## Explanation

Sample full error logs for all checks:

**RegexPatternValidation Check:**

File name/path violation: <failure_message>
- file_path_1
- file_path_2

filePath1:lineNumber1: errorToShow1
filePath2:lineNumber2: errorToShow1
filePath3:lineNumber3: errorToShow2

Refer to <wiki_url_1> for more details on how to fix this.

**XmlSyntaxValidation Check:**

errorFile1:lineNumber1:columnNumber1: error_message1
errorFile2:lineNumber2:columnNumber2: error_message2
errorFile3:lineNumber3:columnNumber3: error_message3

Refer to <wiki_url_2> for more details on how to fix this.

**TestFilePresence Check:**

File file_path_1 does not have a corresponding test file.
File file_path_2 does not have a corresponding test file.
File file_path_3 does not have a corresponding test file.

Refer to <wiki_url_3> for more details on how to fix this.

**AccessibilityLabel Check:**

Accessibility label missing for Activities:
- activityPath1
- activityPath2
- activityPath3

Refer to <wiki_url_4> for more details on how to fix this.

**KDocValidity Check:**

KDoc missing for files:
- file_path_1:columnNumber
- file_path_2:columnNumber
- file_path_3:columnNumber
- file_path_4:columnNumber

Refer to <wiki_url_5> for more details on how to fix this.

**TodoOpen Check:**

TODOs not in proper format:
- file_path_1:lineNumber1
- file_path_2:lineNumber2
- file_path_3:lineNumber3

TODOs not corresponding to open issues on GitHub:
- file_path_4:lineNumber4
- file_path_5:lineNumber5
- file_path_6:lineNumber6

Refer to <wiki_url_6> for more details on how to fix this.

**TodoIssueResolved Check:**

The following TODOs are unresolved for the closed issue:
- file_path_4:lineNumber4
- file_path_5:lineNumber5
- file_path_6:lineNumber6

Refer to <wiki_url_7> for more details on how to fix this.
## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt]: We should set up this error such that we only need to print this once. Ditto for other checks.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @Sparsh1212. Took a first pass. Will review fully once the wiki is in place.

I think the approach looks good. Could you please include sample output in the opening comment for each check's full failure case (e.g. all error types triggered)?

File: scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt
Comment (Sparsh1212) [reply scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt]: Done.

**Comment by Sparsh1212** [comment]
> Thanks @Sparsh1212. Took a first pass. Will review fully once the wiki is in place.
> 
> I think the approach looks good. Could you please include sample output in the opening comment for each check's full failure case (e.g. all error types triggered)?

Done. PTAL.

File: scripts/src/javatests/org/oppia/android/scripts/testfile/TestFileCheckTest.kt
Comment (BenHenning) [inline scripts/src/javatests/org/oppia/android/scripts/testfile/TestFileCheckTest.kt]: Shouldn't we also be checking that it's at the end of the output for this & the other test file checks?

File: scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxCheckTest.kt
Comment (BenHenning) [inline scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxCheckTest.kt]: Ditto for this & the next test.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @Sparsh1212. Fully reviewed this--just had 2 comments. I still can't approve until the wiki page is checked in, but the PR looks pretty close to done.

**Comment by BenHenning** [comment]
Also, @Sparsh1212 please fix the CI checks & assign this back once the base PR is merged.

File: scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxCheckTest.kt
Comment (Sparsh1212) [reply scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxCheckTest.kt]: Done.

File: scripts/src/javatests/org/oppia/android/scripts/testfile/TestFileCheckTest.kt
Comment (Sparsh1212) [reply scripts/src/javatests/org/oppia/android/scripts/testfile/TestFileCheckTest.kt]: Done.

File: scripts/src/java/org/oppia/android/scripts/label/AccessibilityLabelCheck.kt
Comment (Sparsh1212) [inline scripts/src/java/org/oppia/android/scripts/label/AccessibilityLabelCheck.kt:25]: Missed Typo fixed now.

**Review by BenHenning — APPROVED** [review]
Thanks @Sparsh1212. LGTM! Just waiting for CI checks to pass before merging.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Looks like there's 1 CI failure. PTAL.

**Comment by vinitamurthi** [comment]
@BenHenning I think all checks are passing here right?

**Review by BenHenning — APPROVED** [review]
Thanks @Sparsh1212. LGTM. Nice find regarding the flake.

**Pull request merged**
